### PR TITLE
refactor(wakunode2): encapsulate wakunode2 setup and fix warnings

### PR DESCRIPTION
**Pull request by LNSD**
This is a small gardening PR 🪴 

~~- [x] Fix several warnings about the `except` clause.~~
- [x] Wrap the Wakunode2 setup and main logic in two procedures.

### CODE REVIEW

**Review by rymnc — APPROVED** [review]
LGTM

**Comment by LNSD** [comment]
Putting the PR back to draft due to the following compilation error:
```
/home/lorenzo/code/status/nwaku/apps/wakunode2/wakunode2.nim(744, 29) Error: illegal capture 'node' because 'handleCtrlC' has the calling convention: <noconv>
```

File: apps/wakunode2/wakunode2.nim [hunk 1]
```diff
@@ -410,16 +410,9 @@ proc setupProtocols(node: WakuNode, conf: WakuNodeConf,
 
       try:
         await node.mountRlnRelay(rlnConf)
-      except:
+      except CatchableError:
         return err("failed to mount waku RLN relay protocol: " & getCurrentExceptionMsg())
 
-  if conf.swap:
```
Comment (alrevuelta) [inline apps/wakunode2/wakunode2.nim:416]: should we also remove the flag?
https://github.com/waku-org/nwaku/blob/master/apps/wakunode2/config.nim#L261

**Comment by LNSD** [comment]
Closing this PR. I am starting the refactoring from the `waku_node` module.

**Pull request closed without merging**